server/test/services/apikey/ApiKeyServiceTest.java

```diff
@@ -297,11 +297,18 @@ public void retireApiKey_keyAlreadyRetired_throws() {
                         "grant-program-read[test-program]", "true")),
                 adminProfile)
             .getApiKey();
-    apiKeyService.retireApiKey(apiKey.id, adminProfile);
 
-    assertThatThrownBy(() -> apiKeyService.retireApiKey(apiKey.id, adminProfile))
+    assertThat(apiKey.getRetiredTime().isPresent())
+        .withFailMessage("Key is retired, but should not be.")
+        .isFalse();
+    ApiKeyModel retiredApiKey = apiKeyService.retireApiKey(apiKey.id, adminProfile);
+    assertThat(retiredApiKey.getRetiredTime().isPresent())
+        .withFailMessage("Key is not retired, but should be.")
+        .isTrue();
+
+    assertThatThrownBy(() -> apiKeyService.retireApiKey(retiredApiKey.id, adminProfile))
```

Comment:
This test never really worked right. It should have always been checking the return object.